rules/rules-overridden-azure/technology-usage/apm.windup.xml

```diff
@@ -0,0 +1,115 @@
+        </description>
+        <dependencies>
+          <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final"/>
+          <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final"/>
+        </dependencies>
+        <targetTechnology id="azure-spring-apps"/>
+        <targetTechnology id="azure-appservice"/>
+        <targetTechnology id="azure-aks"/>
+        <targetTechnology id="azure-container-apps"/>
+        <tag>apm</tag>
+        <overrideRules>true</overrideRules>
+    </metadata>
+    <rules>
+        <rule id="apm-00000">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="com.microsoft.azure" artifactId="{*}applicationinsights{*}"/>
+                    </project>
+                    <file filename="applicationinsights{*}.jar"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Application Performance Management (APM) tool - Application Insights" category-id="mandatory" effort="1">
+                    <message>
+                        To migrate a Java application that uses Application Insights to Azure, you can follow this recommendation:
+
+                        * **Configure Application Insights**: To connect Application Insights with Java applications deployed on Azure, you can turn on Application Insights in the portal page of your app or configure it through Azure CLI. For a detailed guide, please refer to the following documentation links.
+                    </message>
+                    <link title="Configure Application Insights to connect with Azure App Service" href="https://learn.microsoft.com/azure/app-service/configure-language-java#configure-application-insights"/>
+                    <link title="Configure Application Insights to connect with Azure Spring Apps" href="https://learn.microsoft.com/azure/spring-apps/how-to-application-insights#manage-application-insights-using-the-azure-portal"/>
+                    <link title="Application Insights" href="https://learn.microsoft.com/azure/azure-monitor/app/app-insights-overview"/>
+                </hint>
+                <technology-tag level="INFORMATIONAL">Application Insights</technology-tag>
+            </perform>
+        </rule>
+        <rule id="apm-00001">
+            <when>
+                <or>
+                    <project>
+                        <artifact groupId="{*}newrelic{*}" artifactId="{*}"/>
+                    </project>
+                    <file filename="newrelic{*}.jar"/>
+                </or>
+            </when>
+            <perform>
+                <hint title="Application Performance Management (APM) tool - New Relic" category-id="mandatory" effort="1">
+                    <message>
+                        To migrate a Java application that uses New Relic to Azure, you can follow this recommendation:
```

Comment:
If the application already has New Relic configured, do they really need to take action?

There may be a difference between "adding APM feature to an app" vs "moving APM-enabled app to the Cloud".

Reply:
> If the application already has New Relic configured, do they really need to take action?
> 
> There may be a difference between "adding APM feature to an app" vs "moving APM-enabled app to the Cloud".

@brunoborges When moving a New Relic-enabled app to App Service, customers need to [upload the New Relic Java agent files](https://learn.microsoft.com/en-us/azure/app-service/configure-language-java?pivots=platform-linux#configure-new-relic) so that the javaagent option can work. (For other Azure services, the integration method may vary. For example, with Azure Spring Apps, the New Relic agent file is provided and customers need to configure the jvm-options and environment variables as instructed.)

Reply:
Ok, that in this case the rule is `mandatory`, right ?